### ISSUE TITLE
only the first disk is passed to bhyveload

### ISSUE THREAD

**Issue by avg-I** [churchers/vm-bhyve#470]
It looks like vm-bhyve passes only the first configured disk to bhyveload via its -d option.
Is there a way to pass all disks to bhyveload?

I think that the current / default behavior makes sense for the typilcal VM configurations.
But passing all disks can be useful for things like booting from ZFS mirror or raidz.

As a side question, is it possible to boot FreeBSD guests with loader="uefi-cms" ?

Thank you.

**Comment by churchers** [comment]
Yes you should be able to use `uefi-csm` as a loader option, although I'm never had much luck with it.

I can't find any info on passing multiple disks to bhyveload? The man page just lists it as the option for the boot disk. I may be completely wrong but I was always under the impression that the initial boot loader was read from a single disk? In the case of real ZFS pools you'd just put boot code on every disk so it can boot from the first one it finds...

**Comment by avg-I** [comment]
-d option can be used multiple times and each disk is passed to bhyveload.
When the loader runs it sees only disks that BIOS (on real hardware) or bhyveload (under bhyve) provides.
The simplest boot stage like gptzfsboot is read from a raw partition, but other ZFS pool data needs all disks that constitute a ZFS pool.

**Comment by p4tpr0** [comment]
I came here with the exact same problem but I found a solution. The code of vm bhyve includes a reference to a bhyveload_args variable, so I’ve added this in my vm config file and it worked perfectly:

`bhyveload_args="-d /sas/vm/freebsd13-zraid1-refactor/disk1.img -d /sas/vm/freebsd13-zraid1-refactor/disk2.img -d /sas/vm/freebsd13-zraid1-refactor/disk3.img"`

disk0.img is added by default and bhyveload_args makes it possible to add more disks (here a 4 disks Zraid1)
